opengovsg/postmangovsg
# fix: display error page if campaign id is invalid; improve layout

## Problem

The UI is stuck in the loading state whenever the user navigates to an invalid campaign page (/campaigns/<invalid campaign id>).

Closes #1022 

## Solution

**Features**
- Navigate to the error page if the campaign ID is invalid
- Center-align text on the error page
- Add navigation buttons to the previous page and to the dashboard to the error page

**Screnshots**
<img width="1536" alt="Screenshot 2021-03-22 at 10 13 26 AM" src="https://user-images.githubusercontent.com/4538946/111931954-54407e00-8af7-11eb-8d6b-6b69e08a4eff.png">
<img width="357" alt="Screenshot 2021-03-22 at 10 13 45 AM" src="https://user-images.githubusercontent.com/4538946/111931959-5571ab00-8af7-11eb-926d-7f40005ff18e.png">



## Tests

1. Deploy the app.
2. Navigate to an invalid campaign URL, such as /campaigns/12378123461248124 (does not exist) or /campaigns/notapositiveinteger (not a positive integer).
3. The frontend should redirect to the main campaigns dashboard: /campaigns
